sdk-core/src/test/java/care/data4life/sdk/auth/UserServiceTest.kt

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2021 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+
+package care.data4life.sdk.auth
+
+import care.data4life.auth.AuthorizationService
+import care.data4life.crypto.GCKey
+import care.data4life.crypto.GCKeyPair
+import care.data4life.crypto.GCSymmetricKey
+import care.data4life.sdk.ApiService
+import care.data4life.sdk.CryptoSecureStore
+import care.data4life.sdk.CryptoService
+import care.data4life.sdk.log.Log
+import care.data4life.sdk.network.model.EncryptedKey
+import care.data4life.sdk.network.model.UserInfo
+import care.data4life.sdk.network.model.VersionList
+import io.mockk.Called
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.spyk
+import io.mockk.verify
+import io.reactivex.Completable
+import io.reactivex.Single
+import org.junit.Before
+import org.junit.Ignore
+import org.junit.Test
+
+
+class UserServiceTest {
+
+    companion object {
+        private const val AUTH_STATE = "authState"
+        private const val USER_ALIAS = "userAlias"
+        private const val KEY_USER_ID = "user_id"
+        private const val AUTH_TOKEN = "authToken"
+    }
+
+    private lateinit var apiService: ApiService
+    private lateinit var storage: CryptoSecureStore
+    private lateinit var userService: UserService
+    private lateinit var cryptoService: CryptoService
+    private lateinit var authService: AuthorizationService
+
+    @Before
+    fun setUp() {
+        authService = mockk<AuthorizationService>()
+        userService = mockk<UserService>()
+        storage = mockk<CryptoSecureStore>()
+        apiService = mockk<ApiService>()
```

Comment:
You do not need to add the type here on the mockk, it can derive it automatically.